base/src/main/java/io/spine/protobuf/FieldPaths.java

```diff
@@ -99,4 +104,57 @@ public static Object fieldAt(Message holder, FieldPath path) {
         }
         return currentValue;
     }
+
+    /**
+     * Obtains the class of the field at the given field path from the given field holder type.
+     *
+     * @param holderType
+     *         the type of the message to search
+     * @param path
+     *         the field path to search by
+     * @return the class of the requested field
+     */
+    public static Class<?> typeOfFieldAt(Class<? extends Message> holderType, FieldPath path) {
+        checkNotNull(holderType);
+        checkNotNull(path);
+        checkNotEmpty(path);
+
+        Descriptor descriptor = TypeName.of(holderType).getMessageDescriptor();
+        for (Iterator<String> iterator = path.getFieldNameList().iterator(); iterator.hasNext(); ) {
+            String fieldName = iterator.next();
+            FieldDescriptor field = getField(descriptor, fieldName);
+            if (iterator.hasNext()) {
+                checkArgument(field.getType() == MESSAGE,
+                              "Field %s of type %s is not a message field.");
+                descriptor = field.getMessageType();
+            } else {
+                return classOf(field);
+            }
+        }
+        throw new IllegalStateException("Unreachable statement.");
```

Comment:
Please describe what's wrong instead of just giving this message.

Besides, I think that `Exceptions` is worth usage in this case.